### PR TITLE
Add test for file cleanup after extraction

### PR DESCRIPTION
**Pull request by cameronmarsh**

### CODE REVIEW

**Comment by bbalser** [comment]
If an extractions errors in the middle of processing the stream, I believe the file will not be removed correctly.  If true, we should handle this.

File: apps/service_gather/test/gather/extraction_test.exs [hunk 1]
```diff
@@ -174,4 +180,43 @@ defmodule Gather.ExtractionTest do
...
+    Gather.WriterMock
+    |> expect(:start_link, 1, fn _ -> Agent.start_link(fn -> :dummy end) end)
+
+    request_url = "http://example/download/path"
+
+    allow(Downloader.download(request_url, to: @download_file, headers: []),
+      return: write_temp_file(@download_file)
+    )
+
+    allow(Temp.path(any()), return: {:ok, @download_file})
```
[the remark this answers is not in the document]
Comment (ManApart) [reply apps/service_gather/test/gather/extraction_test.exs]: image

File: apps/service_gather/test/gather/extraction_test.exs [hunk 2]
```diff
@@ -174,4 +180,43 @@ defmodule Gather.ExtractionTest do
...
+        ],
+        dictionary: [
+          Dictionary.Type.Integer.new!(name: "one")
+        ]
+      )
+
+    {:ok, _} = Extraction.start_link(extract: extract)
+
+    assert_async sleep: 1000 do
+      assert !File.exists?(@download_file)
```
[the remark this answers is not in the document]
Comment (ManApart) [reply apps/service_gather/test/gather/extraction_test.exs]: image

**Comment by jdenen** [comment]
Closing in favor of an upcoming PR. It'll use your commit, but I'm rebasing off master and addressing this PR feedback.

**Pull request closed without merging**